From 7ceda7f070a760e1ac23304d0d5e12a0c8d24eed Mon Sep 17 00:00:00 2001
From: Connor McAdams <cmcadams@codeweavers.com>
Date: Wed, 4 Mar 2026 13:03:03 -0500
Subject: [PATCH 16/47] dinput/tests: Test that guidInstance is stable across
 hotplugs.

Signed-off-by: Connor McAdams <cmcadams@codeweavers.com>
(cherry picked from commit c10fe6a4482b87107a52d6be97d6a983953f37aa)

CW-Bug-Id: #25883
---
 dlls/dinput/tests/hotplug.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/dlls/dinput/tests/hotplug.c b/dlls/dinput/tests/hotplug.c
index 9bea45202ff..10a3051e8ce 100644
--- a/dlls/dinput/tests/hotplug.c
+++ b/dlls/dinput/tests/hotplug.c
@@ -154,9 +154,9 @@ static BOOL test_input_lost( DWORD version )
         .dwData = UINT_MAX,
     };
 
-    DIDEVICEINSTANCEW devinst = {.dwSize = sizeof(DIDEVICEINSTANCEW)};
+    DIDEVICEINSTANCEW devinst = {.dwSize = sizeof(DIDEVICEINSTANCEW)}, devinst2 = devinst;
+    IDirectInputDevice8W *device = NULL, *device2;
     DIDEVICEOBJECTDATA objdata[32] = {{0}};
-    IDirectInputDevice8W *device = NULL;
     ULONG ref, count, size;
     DIJOYSTATE2 state;
     HRESULT hr;
@@ -218,6 +218,15 @@ static BOOL test_input_lost( DWORD version )
     ref = IDirectInputDevice8_Release( device );
     ok( ref == 0, "Release returned %ld\n", ref );
 
+    /* Test guidInstance across hotplugs. It should remain the same. */
+    hr = dinput_test_create_device( version, &devinst2, &device2 );
+    ok( hr == DI_OK, "Unexpected hr %#lx.\n", hr );
+    ok( !!device2, "device2 is NULL.\n" );
+    todo_wine ok( !memcmp( &devinst.guidInstance, &devinst2.guidInstance, sizeof(GUID) ),
+                  "Unexpected guidInstance.\n" );
+    ref = IDirectInputDevice8_Release( device2 );
+    ok( ref == 0, "Release returned %ld\n", ref );
+
 done:
     hid_device_stop( &desc, 1 );
     cleanup_registry_keys();
-- 
2.54.0

